### ISSUE TITLE
Small (I hope) change in Prepare > Column: Text > Replace All dialogue

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#4892]
Could we make the single receiver into a multiple receiver?
When it is a multiple receiver we hide or disable to saving option.
Then it saves back into the same columns.

I think saving multiple columns into the same worksheet is not too difficult.  If not it is needed elsewhere as well, so I hope it can be added?  (Perhaps it is similar - or the same as when we save both the residuals and fitted values after fitting a model?)

I address to Danny first, largely to check that this is a simple change.

**Comment by Muthenya** [comment]
@dannyparsons what's your take on this?

**Comment by dannyparsons** [comment]
Yes, `Replace` and `Replace All` could have a multiple receiver. There should then be an option (radio buttons) of either replacing the original columns, or providing a prefix for the new columns.

**Comment by Muthenya** [comment]
Okay. @dannyparsons do you have an idea of how the dialog should appear now?

**Comment by dannyparsons** [comment]
Multiple receiver, then instead of "new column name" two radio buttons for replace columns or new columns and if new columns a text box for the prefix for the new columns.

**Comment by Ivanluv** [comment]
Is this ready to be worked on?

**Comment by lilyclements** [comment]
@rdstern this is interesting and very similar to proposed changes for #3751: We talk here about having a multiple receiver, but I wonder if we now instead want an option to use our special "select" option.

**If so then we want:** 

* This is the "Prepare > Column: Text > Find/Replace" dialog
* Under the "Replace" tab we want to have a set of radio buttons above the column selector (like in our "Describe > Column: Numeric > Transform dialog" dialog)

I've made an example here with the survey data, and I set the three factors (village, variety, fertgrp) to be in a select called "selection".

**If the "Column" is chosen, then the R code doesn't change from what it currently is - at the moment, it is this:**

```r
village <- data_book$get_columns_from_data(data_name="survey", col_names="village", use_current_filter=FALSE)
replace_all <- stringr::str_replace_all(string=village, stringr::coll("B", FALSE), replacement="_hello_")
data_book$add_columns_to_data(data_name="survey", col_name="replace_all", col_data=replace_all, before=FALSE, adjacent_column="village")
rm(list=c("replace_all", "village"))
```

**If the "Select" is chosen, then the R code is:**

```r
# first a quick bit of set up to add a column selection to all factors
data_book$add_column_selection(data_name="survey", name="selection", column_selection=list(C0=list(operation="tidyselect::where", parameters=list(fn=is.factor))), and_or="|")

survey <- data_book$get_data_frame("survey", column_selection_name ="selection")
new_name <- "NEWNAME" # our new column name (suffix)
col <- survey %>%
  dplyr::mutate(dplyr::across(everything(.), ~stringr::str_replace_all(string=.x, pattern = stringr::coll("B", FALSE), replacement="_hello_") %>%
                                as.data.frame(), 
                              .names = paste0("{.col}_", new_name))) %>%
                              tidyr::unpack(cols = ends_with(paste0("_", new_name)), names_sep = ".")
data_book$add_columns_to_data(data_name="survey", col_data=col, before=FALSE)
rm(list=c("col", "survey"))
```

If we're happy then @Ag-Derek can you assign someone to make these changes here? This should be reasonably straightforward, I hope, but might need some support if taken on by a newer programmer. These changes are virtually the same here as they are in the proposed changes in #3751

**Comment by rdstern** [comment]
@lilyclements and @derekagorhom I like this a lot.  There's a set of dialogs (in the Prepare menu) that we usually use for a single variable.  Hence a single receiver.  But sometimes we want to apply the same function to a whole set of columns.  We have this already fully in place (I think) in the Prepare > Column: Numeric > Transform dialog.  

I also really like the idea that this is a good use of our select facility.  So, let's proceed.  @derekagorhom I wonder if you are best placed to check the code and existing system in Transform and then include it here.  Then including it in more dialogs can be handed on to one of the others?
